src/pydrex/minerals.py

```diff
@@ -358,8 +362,7 @@ def apply_gbs(orientations, fractions, config):
                 len(np.nonzero(mask)) / len(fractions),
             )
             # No rotation: carry over previous orientations.
-            # TODO: Should we really be resetting to initial orientations here?
-            orientations[mask, :, :] = self.orientations[0][mask, :, :]
+            orientations[mask, :, :] = self.orientations[-1][mask, :, :]
```

Comment:
This changes from FORTRAN D-Rex, right?

Reply:
Not from what I can tell, at least from the version on the website (not sure about Chris' or Navid's version). In the original one, the sliding parameter (chi) only appears 4 times in the code, on the lines 66 (declaration), 388 (read value from input file), 1385 and 1460 (sliding for olivine and enstatite, respectively). In the last two cases, the context is identical (just swapping dotacs -> dotacs_ens and rt -> rt_ent).

```
!!! grain boundary sliding for small grains
   IF (odfi(i) .LT. chi/REAL(size)) THEN
      dotacs(i,:,:) = 0d0
      rt(i) = 0d0
   END IF
```

In other words, they set the differential of the orientation to zero, not the orientation itself (and also the dislocation density `rt`). In our code that would be the orientations_diff, not orientations. Setting that to zero is equivalent to carrying over the previous orientation as I have done here.

Reply:
Fair. I cannot quite find the updated FORTRAN file, but I am pretty sure they changed that bit. Have you checked what is implemented in ASPECT?

Reply:
It makes no sense to reset to index 0 at all, what if the initial condition is not isotropic?

Aspect implementation checks the greater than condition instead, and if the check fails then they simply set the strain energy to zero and leave the orientations unchanged which is like using index [-1] here not [0]:

https://github.com/geodynamics/aspect/blob/57d11548c089c3d44b40a84714e8c8f0e1938b3e/source/particle/property/crystal_preferred_orientation.cc#L781-L795